cmd/agent/app/config.go

```diff
@@ -212,14 +216,24 @@ func (b *Builder) CreateAgent(mFactory metrics.Factory, logger *zap.Logger) (*Ag
 		b.CollectorServiceName = defaultCollectorServiceName
 	}
 
+	// Combine CollectorHostPort and CollectorHostPorts to
+	// form a FixedDiscoverer if applicable.
+	var hostPorts []string
+	hostPorts = append(hostPorts, b.CollectorHostPorts...)
+	if b.CollectorHostPort != "" {
+		hostPorts = strings.Split(b.CollectorHostPort, ",")
+	}
+	if len(hostPorts) != 0 {
+		d := discovery.FixedDiscoverer(hostPorts)
+		notifier := &discovery.Dispatcher{}
+		b.WithDiscoverer(d).WithDiscoveryNotifier(notifier)
+	}
```

Comment:
yeah, I am not very happy with the duplication myself. The builder struct is designed to be yaml-friendly. We internally have a different main.go for the agent that includes some additional internal components, and a yaml-file-based configuration. And in yaml it's a lot nicer to define the addresses as a list
```
   - collectorHostPorts:
     - hostA:57000
     - hostB:57000
     - hostC:57000
```
rather than comma-separated string
```
   - collectorHostPort: hostA:57000,hostB:57000,hostC:57000
```

My preferred model would be to have only `CollectorHostPorts []string`. But I don't know how to bind it to a cmd line flag nicely.

If you want to drop the `[]string` var and only keep the comma-separated string, I am fine with it.